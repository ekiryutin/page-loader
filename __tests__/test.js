import nock from 'nock';
import fs from 'fs';
import path from 'path';
import os from 'os';
import downloadPage from '../src';

const expectedFile = path.resolve(__dirname, '__fixtures__/test.html');

const server = 'https://host';
const page = '/test';
const testUrl = `${server}${page}`;

let outputDir;

beforeAll(async () => {
  outputDir = await fs.promises.mkdtemp(path.join(os.tmpdir(), path.sep));
});

test(`download ${page}`, async () => {
  nock(server)
    .get(page)
    .replyWithFile(
      200,
      expectedFile,
      { 'Content-Type': 'text/html' },
    );
  await downloadPage(testUrl, outputDir);

  const expected = await fs.promises.readFile(expectedFile);
  const actual = await fs.promises.readFile(`${outputDir}/host-test.html`);

  // expect(actual).toBe(expected);
  expect(actual.equals(expected)).toBe(true);
});
