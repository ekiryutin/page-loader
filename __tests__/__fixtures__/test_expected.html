<!DOCTYPE html><html lang="ru"><head>
  <meta charset="UTF-8">
  <title>Тест</title>
  <link rel="stylesheet" href="host-test_files/css-style.css">
  <link rel="stylesheet" href="//cdn.bootstrap.com/superstyle.css">
  <link rel="stylesheet" href="host-test_files/css-invalid.css">
</head>
<body>
  <h1>Заголовок</h1>
  <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit, 
    sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. 
    Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. 
    Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. 
    Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum. 
  </p>
  <img src="host-test_files/img-test.jpg">
  <hr>
  <a href="https://yandex.ru">Внешняя ссылка</a>
  <script src="host-test_files/js-script.js"></script>

</body></html>